testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/CIBATest.java

```diff
@@ -0,0 +1,1196 @@
+    }
+
+    private String cibaBackchannelTokenDeliveryMode;
+    private Integer cibaExpiresIn;
+    private Integer cibaInterval;
+    private String cibaAuthRequestedUserHint;
+
+    private final String TEST_REALM_NAME = "test";
+    private final String TEST_CLIENT_NAME = "test-app";
+    private final String TEST_CLIENT_PASSWORD = "password";
+
+    private AuthenticationRequestAcknowledgement doBackchannelAuthenticationRequest(String username) throws Exception {
+        return doBackchannelAuthenticationRequest(TEST_CLIENT_NAME, TEST_CLIENT_PASSWORD, username, null);
+    }
+
+    private AuthenticationRequestAcknowledgement doBackchannelAuthenticationRequest(String clientId, String clientSecret, String username, String bindingMessage) throws Exception {
+        AuthenticationRequestAcknowledgement response = oauth.doBackchannelAuthenticationRequest(clientId, clientSecret, username, bindingMessage);
+        Assert.assertThat(response.getStatusCode(), is(equalTo(200)));
+        Assert.assertNotNull(response.getAuthReqId());
+        return response;
+    }
+
+    private AuthenticationChannelRequest doAuthenticationChannelRequest() {
+        // get Authentication Channel Request keycloak has done on Backchannel Authentication Endpoint from the FIFO queue of testing Authentication Channel Request API
+        TestOIDCEndpointsApplicationResource oidcClientEndpointsResource = testingClient.testApp().oidcClientEndpoints();
+        AuthenticationChannelRequest authenticationChannelReq = oidcClientEndpointsResource.getAuthenticationChannel();
+        return authenticationChannelReq;
+    }
+
+    private EventRepresentation doAuthenticationChannelCallback(AuthenticationChannelRequest authenticationChannelReq, String authenticationChannelStatus, String username) throws Exception {
+        return doAuthenticationChannelCallback(TEST_CLIENT_NAME, authenticationChannelReq, authenticationChannelStatus, username);
+    }
+
+    private EventRepresentation doAuthenticationChannelCallback(String clientIdAsConsumerDevice, AuthenticationChannelRequest authenticationChannelReq, String authenticationChannelStatus, String username) throws Exception {
+        int statusCode = oauth.doAuthenticationChannelCallback(AUTHENTICATION_CHANNEL_SERVER_NAME, AUTHENTICATION_CHANNEL_SERVER_PASSWORD, authenticationChannelReq.getUserInfo(), authenticationChannelReq.getAuthenticationChannelId(), authenticationChannelStatus);
+        Assert.assertThat(statusCode, is(equalTo(200)));
+        // check login event : ignore user id and other details except for username
+        EventRepresentation representation = new EventRepresentation();
+
+        representation.setDetails(Collections.emptyMap());
+
+        return representation;
+    }
+
+    private EventRepresentation doAuthenticationChannelCallback(String clientId, AuthenticationChannelRequest authenticationChannelReq, String authenticationChannelStatus, String username, String error) throws Exception {
```

Comment:
It seems this method is not used and can be removed

Reply:
@mposolda Yes, I will remove it.

Reply:
@mposolda I've already removed this method.